.palantir/revapi.yml

```diff
@@ -456,6 +456,28 @@ acceptedBreaks:
       new: "parameter void org.apache.iceberg.actions.BaseCommitService<T>::offer(===T===)\
         \ @ org.apache.iceberg.actions.RewriteDataFilesCommitManager.CommitService"
       justification: "Backwards compatible parameterization of argument"
+    - code: "java.method.parameterTypeChanged"
+      old: "parameter void org.apache.iceberg.rest.RESTCatalog::<init>(===java.util.function.Function<java.util.Map<java.lang.String,\
+        \ java.lang.String>, org.apache.iceberg.rest.RESTClient>===)"
+      new: "parameter void org.apache.iceberg.rest.RESTCatalog::<init>(===org.apache.iceberg.util.SerializableFunction<java.util.Map<java.lang.String,\
+        \ java.lang.String>, org.apache.iceberg.rest.RESTClient>===)"
+      justification: "Switching to SerializableFunction"
```

Comment:
Can we support both? I don't think we can make this change if it is a breaking one.

Reply:
I don't think we can support both because there's no place where we could just cast from `Function` to `SerializableFunction`

Reply:
@rdblue if we don't want to break the API with the `Function` -> `SerializableFunction` change, then the other alternative would be to make `HTTPClient` serializable. I've opened https://github.com/apache/iceberg/pull/8032 to do that